completion_aggregator/migrations/0004_index_stalecompletion.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.10.8 on 2018-10-18 00:13
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+import django.utils.timezone
+import model_utils.fields
+import opaque_keys.edx.django.models
+import six
+
+
+BATCH_SIZE = 1000
+
+
+def copy_data(apps, schema_editor):
+    """
+    Add indexes to completion_aggregator_stalecompletion without locking.
+
+    Copy data into the new stalecompletionnew table in batches, then rename
+    both tables, then do a final copy of any new data that came in.
+    """
+
+    copy_sql_tmpl = """
+        INSERT INTO %(target)s (`created`, `modified`, `username`, `course_key`, `block_key`, `force`, `resolved`)
+            SELECT `created`, `modified`, `username`, `course_key`, `block_key`, `force`, `resolved`
+                FROM %(source)s
+                WHERE id > %%s
+                ORDER BY id
+                LIMIT %%s OFFSET %%s
+    """
+    StaleCompletion = apps.get_model("completion_aggregator", "StaleCompletion")
+    cursor = schema_editor.connection.cursor()
+    max_id = StaleCompletion.objects.order_by('-id')[0].id
+    count = StaleCompletion.objects.count()
```

Comment:
If anything gets added before this line can run but after we already have a `max_id`, then when trying to insert any new data that came in while performing the initial copy, we'll end up trying to insert data that already exists in the table, namely data with `id > max_id AND id < count`. Is that going to cause an issue? If so we can run these 2 as one transaction.